sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/ExpressionConverter.java

```diff
@@ -770,88 +765,34 @@ private RexNode convertResolvedFunctionCall(
           throw new UnsupportedOperationException(
               "Only support TUMBLE, HOP AND SESSION functions right now.");
       }
-    } else if (functionCall.getFunction().getGroup().equals("ZetaSQL")) {
-      op =
-          SqlStdOperatorMappingTable.ZETASQL_FUNCTION_TO_CALCITE_SQL_OPERATOR.get(
-              functionCall.getFunction().getName());
-
+    } else if ("ZetaSQL".equals(funGroup)) {
       if (op == null) {
-        throw new UnsupportedOperationException(
-            "Does not support ZetaSQL function: " + functionCall.getFunction().getName());
+        TypeKind returnType =
+            SqlStdOperatorMappingTable.ZETASQL_FUNCTION_NAME_TO_RETURN_TYPE.get(funName);
```

Comment:
Manually maintaining this list seems like no fun. What about using `functionCall.getSignature().getResultType().getType()`?

Reply:
Great suggestion! Removing the manual map simplifies the code quite a lot.